File: Contest/Biweekly/Biweekly Contest 170/3750. Minimum Number of Flips to Reverse Binary String/Solution.py
class Solution:
    def minimumFlips(self, n: int) -> int:
        # N : len of binary representaion of the 'n'
        # Time Complexity: O(N)
        # Space Complexity: O(N)
        binS = bin(n)[2:]
        n = len(binS)

        minFlips = 0
        l, r = 0, n - 1
        while l < r:
            if binS[l] != binS[r]:
                minFlips += 2
            l += 1
            r -= 1
        return minFlips

File: Contest/Biweekly/Biweekly Contest 170/3750. Minimum Number of Flips to Reverse Binary String/Solution.java
import java.util.*;

class Solution {
    public int minimumFlips(int n) {
        // N : len of binary representaion of the 'n'
        // Time Complexity: O(N)
        // Space Complexity: O(N)
        String binS = Integer.toBinaryString(n);
        int m = binS.length();
        int l = 0;
        int r = m - 1;
        int totalFlips = 0;
        while (l < r) {
            if (binS.charAt(l) != binS.charAt(r)) {
                totalFlips += 2;
            }
            l++;
            r--;
        }
        return totalFlips;
    }
}